### PR TITLE
Allow (un)publishing of workflows

### LINKED ISSUE #361: Allow (un)publishing of workflows
Add a `published` param to the `update_workflow` method of the workflow client, so that workflows can be published and unpublished with bioblend. See https://github.com/galaxyproject/galaxy/pull/10748.

### PR DESCRIPTION
**Pull request by rikeshi**
Resolves #361 

This is my first pull request to become familiar with the project.

### CODE REVIEW

File: bioblend/_tests/TestGalaxyWorkflows.py [hunk 1]
```diff
@@ -169,6 +169,15 @@ def test_update_workflow_name(self):
         updated_wf = self.gi.workflows.update_workflow(wf['id'], name=new_name)
         self.assertEqual(updated_wf['name'], new_name)
 
+    @test_util.skip_unless_galaxy('release_18.05')
+    def test_update_workflow_published(self):
+        path = test_util.get_abspath(os.path.join('data', 'paste_columns.ga'))
+        wf = self.gi.workflows.import_workflow_from_local_path(path)
```
Comment (simonbray) [inline bioblend/_tests/TestGalaxyWorkflows.py:175]:
```suggestion
        self.assertFalse(updated_wf['published'])
        wf = self.gi.workflows.import_workflow_from_local_path(path)
```

Comment (simonbray) [reply bioblend/_tests/TestGalaxyWorkflows.py:175]: Just for completeness - the workflow should be unpublished by default.

Thanks for the PR @rikeshi.

File: bioblend/_tests/TestGalaxyWorkflows.py [hunk 2]
```diff
@@ -169,6 +169,15 @@ def test_update_workflow_name(self):
         updated_wf = self.gi.workflows.update_workflow(wf['id'], name=new_name)
         self.assertEqual(updated_wf['name'], new_name)
 
+    @test_util.skip_unless_galaxy('release_18.05')
```
Comment (simonbray) [inline bioblend/_tests/TestGalaxyWorkflows.py:172]:
```suggestion
    @test_util.skip_unless_galaxy('release_21.01')
```

Comment (simonbray) [reply bioblend/_tests/TestGalaxyWorkflows.py:172]: as this feature is only in the galaxy dev branch

**Review by simonbray — APPROVED** [review]

**Comment by simonbray** [comment]
This looks good to me. I am just wondering why tests are not being run. Any idea @nsoranzo?

**Comment by bgruening** [comment]
Travis has currently some problems. We are working on it, or we would need to merge/work on @OlegZharkov PR https://github.com/galaxyproject/bioblend/pull/359

**Comment by bgruening** [comment]
@rikeshi can you please merge master into your branch?

**Comment by simonbray** [comment]
Looks good, thanks @rikeshi!

**Pull request merged**